README.md

```diff
@@ -72,43 +79,69 @@ Now write the tools for exact pattern matching. You can use the naive algorithm
 > diff naive.sam lin.sam
 ```
 
-You might not have to sort the output, if you run through reads
+ANSWER:
+See src/naive.py (naive_algorithm()) and src/lin.py (ba_algorithm())
 
 ## Evaluation
 
 Implement the two algorithms in two tools, `naive` and `lin`, that must be present at the root of the repository once they are built. The test setup checks that they give the correct output on selected data, but you should still carefully test them.
 
 Once you have implemented the tools, fill out the report below. 
 
+ANSWER:
+See src/naive.py and src/lin.py
+
 ## Report
 
 ### Insights you may have had while implementing and comparing the algorithms. 
 
-*Describe this here.*
+ANSWER:
+Finally understood how border-arrays enables efficient/linear string-searching.
 
 ### Problems encountered if any. 
 
-*Describe this here.*
+ANSWER:
+I had some trouble jumping backwards (or rather figuring out that i had to repetitively jump back until i reached match) in my border-array (kind of a brain-fart i know) (finally fixed it with a while-loop). 
 
 ### Experiments that verifies the correctness of your implementations.
 
-*Describe this here.*
+ANSWER:
+I simulate a bunch of random DNA sequences of varying lengths (50-100 nucleotides; representing ref seqs), and from these sequence sampled random intervals of length 1-20 representing exact-read seqs. These samples were then used to test if my implentations returned the exact intervals/positions (compared to re.findall(), see first section of src/runtime_comparison.py) through 500000 iterations. 
+Hereafter i similarily sampled ref/reads while feeding these to both algorithms (naive vs lin) simultaneously. The algorithms returned the same result for all iterations (500000; see src/runtime_comparison.py). A coverage test was performed using coverage.py (https://coverage.readthedocs.io/en/6.4.4/). For this purpuse a fasta file containing 5 sequences and a corresponding simple fastq file containing 100 reads were generated (using scripts provided by teacher in previus week, see data_for_testing/simulate_simple_fasta.py; data_for_testing/simulate_simple_fastq.py). The naive and lin algorithms had respectivly 95% and 94% coverage where all missing lines (naive: 15, 17, 21, 23; lin: 16, 32, 42, 44, 48, 50) was involved in early returnings as response to empty/incompatable inputs. These responses were then tested manually i.e. both algorithms were tested for varius empty input cases (empty/empty and non-empty/empty etc.) Sequences containing first and last 10 nucleotides repeatedly was used to test 'takeoff and landing'. 
```

Comment:
It's always great if you can add a test suite (using e.g. `pytest` or similar) to cover all lines. You cannot always if some of the code handles cases that really should never be reached and should crash the program, but in this application, it makes good sense.